### ISSUE TITLE
Proposal: Closure Templates Change and Release Policy

### ISSUE THREAD

**Issue by Ubehebe** [google/closure-templates#69]
# Motivation
Closure Templates is a mature product with widespread use both inside Google and in the open-source world. In general, we don’t want to change the workflow for writing Closure Templates or the language features available to template authors. Rather, we want to **make it easier to use the best possible Closure Templates features** (and conversely, make it harder or impossible to use Closure Templates in an insecure or non-performant way). We think it is worthwhile to occasionally make breaking changes to Closure Templates in pursuit of this goal.

# Proposed Policy
1. Closure Templates will be released monthly, as close to the start of the month as possible. A release will consist of artifacts published to Maven Central, as well as a release tag in the GitHub repo.
2. We will keep a `BREAKING_CHANGES.md` file in the GitHub repo. This file will be updated every release to describe breaking changes since the last release (if any), and what steps open-source users need to take to upgrade.
3. For every release, we will make an announcement on closure-templates-discuss, pointing to the Maven artifacts, the GitHub tag, the breaking changes, etc.

**Comment by Ubehebe** [comment]
Comments welcome. See [mailing list post](https://groups.google.com/d/msg/closure-templates-discuss/vY-77gj8-F4/0Ez8IF1rBwAJ).

**Comment by hochhaus** [comment]
As a (non-Google) company which heavily uses closure-templates for a proprietary product I think the proposal sounds great. Making the best closure-templates features more accessible while minimizing the likelihood of performance or security issues trumps perfect stability. As breaking changes are documented, the monthly upgrade cycle should not be too hard to stay current with.

**Comment by ChadKillingsworth** [comment]
@Ubehebe If you do that, I'll setup continuous deployment via Travis for publication to npm.

**Comment by nathancomstock** [comment]
This would be excellent and much appreciated !

**Comment by pcj** [comment]
Is there further information to be expected discussing which aspects of Closure Templates are planned to break or change?

**Comment by Ubehebe** [comment]
@pcj We're planning on pushing to GitHub more frequently than once per release. We may add BREAKING_CHANGES metadata to each commit, so before a new release, you can review the commits to get a good idea of what's going to break. 

(Over the past year, pushes have happened about biweekly. We haven't done any pushes since ~November, because we have a few breaking changes in store and I wanted to get a change policy in place before publishing them.)

**Comment by hochhaus** [comment]
@Ubehebe Any idea when the github pushes will resume? I'm itching to experiment with the idom backend.

**Comment by teppeis** [comment]
@Ubehebe @lukesandberg @vrana @emspishak 
Any updates?
`v2018-03-14` was released and it includes hundreds of commits, but I got no release notes, nor `BREAKING_CHANGES.md`.
It's too hard for users outside Google.

Also #97 sync with closure-compiler and closure-library release is better.
CC @ChadKillingsworth

**Comment by lukesandberg** [comment]
@teppeis I agree with you, but unfortunately the team that funds closure templates development at google is understaffed and as such we cannot really support open source users (supporting internal users is our mandate)

I have made an attempt to document this here:
https://github.com/google/closure-templates#support-and-project-status

**Comment by ChadKillingsworth** [comment]
@lukesandberg If you have any external users that you trust, you can allow them rights to publish the NPM version. I'd be happy to give instructions on how that's done.

I think all they really want is for the release to be updated.

**Comment by teppeis** [comment]
@lukesandberg @ChadKillingsworth 
Thanks. This is OSS, I will not make excessive requests.
But all want is please publish release notes and breaking changes if you have for google internal.
Also updating [npm package](https://www.npmjs.com/package/google-closure-templates) would be needed if [closure-webpack-plugin](https://github.com/webpack-contrib/closure-webpack-plugin) supports closure-templates :smile_cat:

**Comment by iteriani** [comment]
Reopening this since I'm now the next maintainer; Anyone know how to update npm packages? :)

**Comment by teppeis** [comment]
related: https://github.com/google/closure-templates/issues/145

**Comment by jplaisted** [comment]
@ChadKillingsworth may know how to update these.

**Comment by ChadKillingsworth** [comment]
I do. I am emailing off thread to assist in getting this set up.

**Comment by banguit** [comment]
Hello,

Any updates regarding npm package releases? Looks like latest npm package is incomplete (broken?) and contains only 3 following files:
- index.js
- package.json
- README.md

**Comment by iteriani** [comment]
I tried to get this set up and it's still a bit confusing for me.

**Comment by banguit** [comment]
Any updates on this?

**Comment by gorohash** [comment]
Is there any update for this? npm package and release notes of Closure Templates are needed in my project.

Thanks for continuing to release Closure Templates!
